### ISSUE TITLE
GitPrePushHook on a new branch with no remote

### ISSUE THREAD

**Issue by ahetawal-p** [pinterest/ktlint#724]
Looking at the current `prepush` script

```git diff --name-only HEAD origin/$(git rev-parse --abbrev-ref HEAD) | grep '\.kt[s"]\?$' | xargs ktlint --relative .```
The script assumes there is a remote present for the current local branch.

But we are seeing unlinted code getting through the pre-push check when a dev is creating a fresh local branch, and trying to push it for the very first time with no remote present. In that case the above doesnt run and all the unlinted code leaks to the repo.

## Expected Behavior
Ktlint prepush hook should trigger on a new local branch with no remote

## Steps to Reproduce
- Create a new local branch, `git checkout -b <local branch>`
- Add some blank lines in the end to trigger the `Needless blank line(s)` error
- Push it to a remote `git push -u origin <local branch>`
- No Ktlint is triggered in this case.

## Your Environment

* Version of ktlint used: `org.jlleitschuh.gradle:ktlint-gradle:9.1.1`

**Comment by Tapchicoma** [comment]
@ahetawal-p I would suggest you to open same issue in [ktlint-gradle](https://github.com/JLLeitschuh/ktlint-gradle) repo. This Git hook is generated by the Gradle plugin and not by ktlint itself.

**Comment by nicolashaan** [comment]
@Tapchicoma it seems that the script is comming from this repo : 
https://github.com/pinterest/ktlint/blob/master/ktlint/src/main/resources/ktlint-git-pre-push-hook.sh
The bug is trivial to reproduce.
